### PR TITLE
peers strength migration to apiv2

### PR DESCRIPTION
**Pull request by morucci**

### CODE REVIEW

File: haskell/src/Monocle/Backend/Queries.hs [hunk 1]
```diff
@@ -107,8 +107,11 @@ changeState state' =
   ]
 
 -- | Add a document type filter to the query
-documentType :: Text -> BH.Query
-documentType type' = BH.TermQuery (BH.Term "type" type') Nothing
+documentType :: [Text] -> BH.Query
+documentType types = BH.TermsQuery "type" $ neList types
+  where
+    neList :: [Text] -> NonEmpty Text
+    neList terms = fromMaybe (error "Terms list cannot be empty") $ nonEmpty terms
```
Comment (TristanCacqueray) [inline haskell/src/Monocle/Backend/Queries.hs:114]:
```suggestion
documentType :: NonEmpty Text -> BH.Query
documentType = BH.TermsQuery "type"
```

File: haskell/src/Monocle/Backend/Queries.hs [hunk 2]
```diff
@@ -326,7 +329,7 @@ getDocTypeTopCountByField :: Text -> Text -> Maybe Int -> QueryFlavor -> QueryM
 getDocTypeTopCountByField doctype attr size qflavor = do
   -- Prepare the query
   basequery <- toBHQueryWithFlavor qflavor <$> getQuery
-  let query = mkAnd $ basequery <> [documentType doctype]
+  let query = mkAnd $ basequery <> [documentType [doctype]]
```
Comment (TristanCacqueray) [inline haskell/src/Monocle/Backend/Queries.hs:332]:
```suggestion
  let query = mkAnd $ basequery <> [documentType $ Data.List.NonEmpty.singleton doctype]
```

Comment (TristanCacqueray) [reply haskell/src/Monocle/Backend/Queries.hs:332]: Or alternatively:  `doctype :| []`

File: haskell/src/Monocle/Backend/Queries.hs [hunk 3]
```diff
@@ -378,51 +378,84 @@ getReposSummary = do
...
+
+getAuthorsPeersStrength :: QueryM [PeerStrengthResult]
+getAuthorsPeersStrength = do
+  authors <-
+    getDocTypeTopCountByField
+      eventTypes
+      "author.muid"
+      (Just 5000)
+      qf
+  authors_peers <- mapM (getAuthorPeers . trTerm) authors
```
Comment (TristanCacqueray) [inline haskell/src/Monocle/Backend/Queries.hs:441]: nit: `mapM` is a legacy, use `traverse` instead.

**Review by TristanCacqueray — APPROVED** [review]

File: haskell/src/Monocle/Backend/Queries.hs [hunk 4]
```diff
@@ -127,8 +131,8 @@ changeState state' =
   ]
 
 -- | Add a document type filter to the query
-documentType :: Text -> BH.Query
-documentType type' = BH.TermQuery (BH.Term "type" type') Nothing
+documentType :: NonEmpty Text -> BH.Query
+documentType = BH.TermsQuery "type"
```
Comment (TristanCacqueray) [inline haskell/src/Monocle/Backend/Queries.hs:135]: Not sure that makes a difference, but we could keep the TermQuery when the list only has a single element.

File: haskell/src/Monocle/Backend/Queries.hs [hunk 5]
```diff
@@ -397,51 +401,100 @@ getReposSummary = do
...
+      "author.muid"
+      (Just 5000)
+      qf
+  authors_peers <- traverse (getAuthorPeers . trTerm) peers
+  pure $
+    take (fromInteger $ toInteger limit) $
+      reverse $
+        sort $
+          filter (\psr -> psrAuthor psr /= psrPeer psr) $
+            concatMap transform authors_peers
```
Comment (TristanCacqueray) [inline haskell/src/Monocle/Backend/Queries.hs:475]: nit: ormolu formats better when using `(.)` composition. e.g.:   `f $ g $ x == f . g $ x`

```suggestion
  pure
    . take (fromInteger $ toInteger limit)
    . reverse
    . sort
    . filter (\psr -> psrAuthor psr /= psrPeer psr)
    $ concatMap transform authors_peers
```

File: haskell/src/Monocle/Backend/Test.hs [hunk 6]
```diff
@@ -370,15 +370,46 @@ testTopAuthors = withTenant doTest
...
-            ]
+
+-- where
+--   indexScenarioNoMerged project cid =
+--     indexScenario
+--       [ s | s <- nominalMerge project cid fakeDate 3600, case s of
+--                                                            SMerge _ -> False
+--                                                            SComment _ -> False
+--                                                            _anyOther -> True
+--       ]
```
Comment (TristanCacqueray) [inline haskell/src/Monocle/Backend/Test.hs:381]:
```suggestion
```

**Review by TristanCacqueray — APPROVED** [review]

File: haskell/src/Monocle/Backend/Queries.hs (hunk 4 above)
Comment (morucci) [reply haskell/src/Monocle/Backend/Queries.hs:135]: It make be the OR for you if needed. I'm in favor of it

**Pull request merged**